### PR TITLE
Add Float16Array and Math.f16round (ES2025)

### PR DESCRIPTION
**Pull request by frostney**
## Summary

Implements IEEE 754 half-precision (binary16) `Float16Array` and `Math.f16round` per the ES2025 Float16Array specification.

- **`Goccia.Float16.pas`** — `DoubleToFloat16` / `Float16ToDouble` conversion helpers with round-to-nearest-even, subnormal handling, and all special values (NaN, ±Infinity, ±0)
- **`Float16Array`** — New typed array kind (`takFloat16`, 2 bytes/element) with full element read/write support, constructor registration, and all shared prototype methods
- **`Math.f16round(x)`** — Rounds a number to the nearest IEEE 754 binary16 value (ES2026 §21.3.2.17)
- Updated `WriteNumberLiteral` and `WriteFloatDirect` to handle Float16 NaN/Infinity correctly
- 10 typed array types total (was 9)

## Testing

- [x] 115 new JS tests passing (both interpreted and bytecode modes):
  - `tests/built-ins/TypedArray/float16array.js` — construction, precision/range, buffer sharing, static methods, prototype methods
  - `tests/built-ins/Math/f16round.js` — special values, rounding, edge cases
  - Updated `tests/built-ins/TypedArray/constructors.js` — Float16Array in global/BYTES_PER_ELEMENT tests
- [x] Full test suite: 4330/4383 passing (12 pre-existing FFI/ASI failures, 0 regressions)
- [x] Documentation updated (`docs/built-ins.md`, `docs/value-system.md`)
- [x] 32 benchmarks in `benchmarks/float16array.js` covering creation, element access, cross-type comparison, methods, buffer sharing, iteration, and Math.f16round
- [x] Formatter passes (`./format.pas --check`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/built-ins.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ See [Adding a New Built-in Type](adding-built-in-types.md) for the complete step
 | `Math.asinh(x)` | Inverse hyperbolic sine |
 | `Math.atanh(x)` | Inverse hyperbolic tangent |
 | `Math.expm1(x)` | e^x - 1 (precise for small x) |
+| `Math.f16round(x)` | Nearest 16-bit float (ES2025) |
```
[the remark this answers is not in the document]
Comment (frostney) [reply docs/built-ins.md]: The implementation refers to ES2026 because that's the living document standard. However, `Math.f16round(x)` was part to ES2025 specification. Agree it's confusing and method names in the built-in documentation do not currently have identifiers on which ES spec they belong to

File: benchmarks/float16array.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    setup: () => {
+      const values = [];
+      for (const i of Array(100).keys()) values.push(i * 1.337);
+      return values;
+    },
+    run: (values) => {
+      for (const v of values) {
+        const r = Math.f16round(v);
+      }
+    },
```
[the remark this answers is not in the document]
Comment (frostney) [reply benchmarks/float16array.js]: Fixed in eb31702 — renamed to `"f16round over 100 values"` as suggested.

File: docs/built-ins.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (frostney) [reply docs/built-ins.md]: Already addressed in 1b2ba46 — removed the edition labels entirely per project owner feedback.

**Comment by frostney** [comment]
Addressed all CodeRabbit review feedback in eb31702:

**🟠 Float-kind checks (major):** Added `takFloat16` to all 5 float-kind branch checks in `TypedArraySort`, `TypedArrayIndexOf`, `TypedArrayLastIndexOf`, and `TypedArrayIncludes`. Without this fix, `Float16Array.sort()` wouldn't move NaN to the end, `includes(NaN)` returned `false`, and `indexOf`/`lastIndexOf` couldn't match stored `±Infinity`.

**Benchmark naming:** Renamed `"f16round vs fround"` → `"f16round over 100 values"` since it only exercises `Math.f16round`.

**Test coverage:** Added two new edge-case tests:
- `indexOf()/lastIndexOf()/includes() handle float special values` — covers NaN (SameValueZero vs strict equality), ±Infinity
- `sort() moves NaN values to the end` — verifies NaN partitioning

All 117 tests pass in both interpreted and bytecode modes. Zero regressions in full suite.
